IndicoToolkit/Association/Positioning.cs

```diff
@@ -0,0 +1,321 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+using IndicoToolkit.Types;
+using IndicoToolkit.Exception;
+
+namespace IndicoToolkit.Association
+{
+    /// <summary>
+    /// Class <c>Positioning</c> helps identify relative positions in a document using bounding box data.
+    /// Positions are expected to contain, at a minimum, the following top-level keys: "bbTop", "bbBot", "bbLeft", "bbRight", "page_num".
+    /// </summary>
+    public class Positioning {
+        
+        public Positioning() { }
+
+        /// <summary>
+        /// Check if the location of one box is above another.
+        /// </summary>
+        /// <param name="abovePos">The position expected to be above</param>
+        /// <param name="belowPos">The position expected to be below</param>
+        /// <param name="mustBeSamePage">Required to be on same page. Defaults to true.</param>
+        public bool positionedAbove(Position abovePos, Position belowPos, bool mustBeSamePage = true)
+        {
+            bool isAbove = false;
+            if (belowPos.pageNum < abovePos.pageNum)
+            {
+                isAbove = false;
+            }
+            else if (belowPos.pageNum != abovePos.pageNum && mustBeSamePage)
+            {
+                isAbove = false;
+            }
+            else if (xAxisOverlap(abovePos, belowPos))
+            {
+                if (belowPos.pageNum == abovePos.pageNum && !yAxisAbove(abovePos, belowPos))
+                {
+                    isAbove = false;
+                }
+                else
+                {
+                    isAbove = true;
+                }
+            }
+            return isAbove;
+        }
+
+        /// <summary>
+        /// Check if the location of one box is on the same page and above another if the lower box's overlap is at least the given percentage.
+        /// </summary>
+        /// <param name="abovePos">The position excepted to be above</param>
+        /// <param name="belowPos">The position excepted to be below</param>
+        /// <param name="minOverlapPercent">The minimum amount of overlap needed. Defaults to null.</param>
+        public bool positionedAboveOverlap(Position abovePos, Position belowPos, float? minOverlapPercent = null)
+        {
+            bool isAbove = false;
+            bool isMinOverlap = true;
+            if (belowPos.pageNum != abovePos.pageNum)
+            {
+                throw new ToolkitInputException(
+                    "Predictions are not on the same page!"
```

Comment:
In the previous function, we just returned false if they weren't on the same page and that was a requirement, but here we throw an exception-- I would make that behavior consistent (one way or the other) unless there is a reason for the two methods to behave differently

Reply:
removed exception and now returning false